templates/c/src/tic80.h

```diff
@@ -0,0 +1,262 @@
+void mouse(MouseData* data);
+
+
+// DRAW / DRAW UTILS
+
+WASM_IMPORT("clip")
+void clip(int32_t x, int32_t y, int32_t w, int32_t h);
+
+WASM_IMPORT("cls")
+void cls(uint8_t color);
+
+WASM_IMPORT("circ")
+void circ(int32_t x, int32_t y, int32_t radius, uint8_t color);
+
+WASM_IMPORT("circb")
+void circb(int32_t x, int32_t y, int32_t radius, uint8_t color);
+
+WASM_IMPORT("elli")
+void elli(int32_t x, int32_t y, int32_t a, int32_t b, uint8_t color);
+
+WASM_IMPORT("ellib")
+void ellib(int32_t x, int32_t y, int32_t a, int32_t b,  uint8_t color);
+
+WASM_IMPORT("fget")
+bool fget(int32_t id, uint8_t flag);
+
+WASM_IMPORT("fset")
+void fset(int32_t id, uint8_t flag, bool value);
+
+WASM_IMPORT("line")
+void line(int32_t x0, int32_t y0, int32_t x1, int32_t y1, uint8_t color);
+
+WASM_IMPORT("map")
+void map(int32_t x, int32_t y, int32_t w, int32_t h, int32_t sx, int32_t sy,
+	  uint8_t* trans_colors, uint8_t colorCount,  int32_t scale,  int32_t remap);
+// todo: why is remap a uint32_t?
+
+WASM_IMPORT("mget")
+uint8_t mget(int32_t x, int32_t y);
+
+WASM_IMPORT("mset")
+void mset(int32_t x, int32_t y, uint8_t value);
+
+WASM_IMPORT("pix")
+uint8_t pix(int32_t x, int32_t y, int8_t color);
+//if color == -1 this should return color
+
+WASM_IMPORT("rect")
+void rect(int32_t x, int32_t y, int32_t w, int32_t h, uint8_t color);
+
+WASM_IMPORT("rectb")
+void rectb(int32_t x, int32_t y, int32_t w, int32_t h, uint8_t color);
+
+WASM_IMPORT("spr")
+void spr(int32_t id, int32_t x, int32_t y, uint8_t* trans_colors, int32_t colorCount,
+	 int32_t scale, int32_t flip, int32_t rotate, int32_t w, int32_t h);
+
+WASM_IMPORT("tri")
+void tri(int32_t x1, int32_t y1, int32_t x2, int32_t y2,
+	 int32_t x3, int32_t y3, uint8_t color);
+
+WASM_IMPORT("trib")
+void trib(int32_t x1, int32_t y1, int32_t x2, int32_t y2,
+	  int32_t x3, int32_t y3, uint8_t color);
+
+WASM_IMPORT("textri")
+void textri(int32_t x1, int32_t y1, int32_t x2, int32_t y2, int32_t x3, int32_t y3,
+	    int32_t u1, int32_t v1, int32_t u2, int32_t v2, int32_t u3, int32_t v3,
+	    bool use_map, uint8_t* trans_colors, int8_t colorCount);
+
+
+// ----
+// TEXT
+
+// print text [x=0 y=0] [color=15] [fixed=false] [scale=1] [smallfont=false] -> text width
+WASM_IMPORT("print")
+int32_t print(const char *text, int32_t x, int32_t y,
+	      int32_t color, bool fixed, int32_t scale, int32_t smallfont);
+// font text, x, y, [transcolor], [char width], [char height], [fixed=false], [scale=1] -> text width
+WASM_IMPORT("font")
+int32_t font(const char *text, uint32_t x, int32_t y, int32_t transcolor,
+	 int32_t char_width, int32_t char_height, bool fixed, int32_t scale);
+
+// -----
+// AUDIO
+
+WASM_IMPORT("music")
+void music(int32_t track, int32_t frame, int32_t row,
+	   bool loop, bool sustain, int32_t tempo, int32_t speed);
+// sfx id [note] [duration=-1] [channel=0]
+//        [volumeleft=15] [volumeright=15] [speed=0]
+WASM_IMPORT("sfx")
+void sfx(int32_t id, int32_t note, int32_t duration, int32_t channel,
```

Comment:
Look again, this API is more complex.  Could be my Zig lib isn't up to date yet.